edfio/edf.py

```diff
@@ -322,6 +322,29 @@ def _set_data(self, data: npt.NDArray[np.float64]) -> None:
         )
         self._digital = np.round(data / gain - offset).astype(np.int16)
 
+    def resample(
+        self, sampling_frequency: float, resampled_data: npt.NDArray[np.float64]
+    ) -> None:
+        """
+        Resample the signal to a new number of samples per data record.
+
+        Parameters
+        ----------
+        samples_per_record : int
+            The new number of samples per data record.
+        resampled_data : npt.NDArray[np.float64]
+            The resampled data. Must not change the signal duration.
+        """
+        factor = sampling_frequency / self.sampling_frequency
+        expected_len = len(self._digital) * factor
+        tolerance = 0.000001
+        if abs(expected_len - len(resampled_data)) > tolerance:
```

Comment:
This might make the validation slightly easier to understand and consistent with the docstring:

```suggestion
        old_duration = len(self._digital) / self.sampling_frequency
        new_duration = len(resampled_data) / sampling_frequency
        tolerance = 0.000001
        if abs(old_duration - new_duration) > tolerance:
```